src/jit/lowerxarch.cpp

```diff
@@ -575,31 +575,6 @@ void Lowering::TreeNodeInfoInit(GenTree* stmt)
             op1 = tree->gtOp.gtOp1;
             op2 = tree->gtOp.gtOp2;
 
-            // See if we have an optimizable power of 2 which will be expanded 
-            // using instructions other than division.
-            // (fgMorph has already done magic number transforms)
-
-            if (op2->IsIntCnsFitsInI32())
-            {
-                bool isSigned = tree->OperGet() == GT_MOD || tree->OperGet() == GT_DIV;
-                ssize_t amount = op2->gtIntConCommon.IconValue();
-
-                if (isPow2(abs(amount)) && (isSigned || amount > 0)
```

Comment:
I would not be in favor of such an assert, as it requires knowing the logic of one phase in another phase, in a way that is not directly communicated.